### PR TITLE
Eliminate Add-Member cmdlet

### PR DESCRIPTION
**Pull request by TrimarcJake**
Add-Member is slow.

### CODE REVIEW

File: Private/Set-AdditionalCAProperty.ps1 [hunk 1]
```diff
@@ -5,7 +5,7 @@
             Mandatory = $true,
             ValueFromPipeline = $true)]
         [array]$ADCSObjects,
-        [System.Management.Automation.PSCredential]$Credential
+        [PSCredential]$Credential
```
Comment (SamErde) [inline Private/Set-AdditionalCAProperty.ps1:8]: Nice, this looks cleaner. Is the full type name implied? Any other benefits or reasons for shortening it?

Comment (TrimarcJake) [reply Private/Set-AdditionalCAProperty.ps1:8]: Uhhhh, you're asking me questions I don't have answers to. Might've been unintentional. Does it still work?

File: Private/Find-AuditingIssue.ps1 [hunk 2]
```diff
@@ -8,26 +8,22 @@
...
-            $Issue | Add-Member -MemberType NoteProperty -Name 'Revert' `
-                -Value "certutil.exe -config $($_.CAFullname) -setreg CA\AuditFilter  $($_.AuditFilter); Invoke-Command -ComputerName `'$($_.dNSHostName)`' -ScriptBlock { Get-Service -Name `'certsvc`' | Restart-Service -Force }" -Force
-            $Issue | Add-Member -MemberType NoteProperty -Name 'Technique' -Value 'DETECT' -Force
+            $Issue['Issue']  = "Auditing is not fully enabled on $($_.CAFullName). Current value is $($_.AuditFilter)"
+            $Issue['Fix']    = "certutil.exe -config `'$($_.CAFullname)`' -setreg `'CA\AuditFilter`' 127; Invoke-Command -ComputerName `'$($_.dNSHostName)`' -ScriptBlock { Get-Service -Name `'certsvc`' | Restart-Service -Force }"
+            $Issue['Revert'] = "certutil.exe -config $($_.CAFullname) -setreg CA\AuditFilter  $($_.AuditFilter); Invoke-Command -ComputerName `'$($_.dNSHostName)`' -ScriptBlock { Get-Service -Name `'certsvc`' | Restart-Service -Force }"
         }
-        $Severity = Set-Severity -Issue $Issue
```
Comment (SamErde) [inline Private/Find-AuditingIssue.ps1:29]: I don't see severity being assigned in the new version. Did you move it?

Comment (TrimarcJake) [reply Private/Find-AuditingIssue.ps1:29]: We never actually used the Severity function, so I removed it.

File: Private/Find-ESC1.ps1 [hunk 3]
```diff
@@ -21,21 +21,17 @@
...
+                    Forest                = $_.CanonicalName.split('/')[0]
+                    Name                  = $_.Name
+                    DistinguishedName     = $_.DistinguishedName
+                    IdentityReference     = $entry.IdentityReference
+                    ActiveDirectoryRights = $entry.ActiveDirectoryRights
+                    Issue = "$($entry.IdentityReference) can enroll in this Client Authentication template using a SAN without Manager Approval"
+                    Fix = "Get-ADObject `'$($_.DistinguishedName)`' | Set-ADObject -Replace @{'msPKI-Certificate-Name-Flag' = 0}"
+                    Revert = "Get-ADObject `'$($_.DistinguishedName)`' | Set-ADObject -Replace @{'msPKI-Certificate-Name-Flag' = 1}"
+                    Technique = 'ESC1'
+                }
```
Comment (SamErde) [inline Private/Find-ESC1.ps1:34]: Severity is missing.

Comment (TrimarcJake) [reply Private/Find-ESC1.ps1:34]: Yep. Intentional!

File: Private/Find-ESC2.ps1 [hunk 4]
```diff
@@ -21,21 +21,17 @@
...
+                    Forest                = $_.CanonicalName.split('/')[0]
+                    Name                  = $_.Name
+                    DistinguishedName     = $_.DistinguishedName
+                    IdentityReference     = $entry.IdentityReference
+                    ActiveDirectoryRights = $entry.ActiveDirectoryRights
+                    Issue                 = "$($entry.IdentityReference) can request a SubCA certificate without Manager Approval"
+                    Fix                   = "Get-ADObject `'$($_.DistinguishedName)`' | Set-ADObject -Replace @{'msPKI-Certificate-Name-Flag' = 0}"
+                    Revert                = "Get-ADObject `'$($_.DistinguishedName)`' | Set-ADObject -Replace @{'msPKI-Certificate-Name-Flag' = 1}"
+                    Technique             = 'ESC2'
+                }
```
Comment (SamErde) [inline Private/Find-ESC2.ps1:34]: Severity is missing.

File: Private/Set-AdditionalCAProperty.ps1 (hunk 1 above)
Comment (TrimarcJake) [reply Private/Set-AdditionalCAProperty.ps1:8]: Yes, [PSCredential] is fine. Woot.

**Comment by TrimarcJake** [comment]
@SamErde if you approve, you have to complete the review with an Approval.

File: Private/Find-ESC8.ps1 [hunk 5]
```diff
@@ -8,26 +8,20 @@
...
                 $Issue['Issue'] = 'HTTPS enrollment is enabled.'
-                $Issue['CAEnrollmentEndpoint'] = $_.CAEnrollmentEndpoint
-                $Issue['Fix'] = 'TBD - Remediate by doing 1, 2, and 3'
-                $Issue['Revert'] = 'TBD'
             }
-            $Issue['Technique'] = 'ESC8'
-            $Severity = Set-Severity -Issue $Issue
-            $Issue['Severity'] = $Severity
-            [PSCustomObject] $Issue
+            [PSCustomObject]$Issue
```
Comment (SamErde) [inline Private/Find-ESC8.ps1:24]: Curious why you added the explicit type for this function, but not the others? (Other than the fact that this existed in the previous version of this function as well.) :)

**Review by SamErde — APPROVED** [review]
Only one minor, non-affecting question on Find-ESC8.

**Pull request merged**